File: Sample/app/src/main/java/net/nend/sample/PagerAndListActivity.java
package net.nend.sample;

import java.util.ArrayList;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;

/**
 * ViewPager内のListViewに表示するサンプル
 */
public class PagerAndListActivity extends FragmentActivity {

    private ViewPager mViewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mViewPager = new ViewPager(this);
        mViewPager.setId(R.id.pager);
        setContentView(mViewPager);

        ArrayList<String> pageList = new ArrayList<String>();
        for (int i = 1; i < 11; i++) {
            pageList.add("page"+i);
        }

        ArrayList<String> itemList = new ArrayList<String>();
        for (int i = 1; i < 51; i++) {
            itemList.add("item"+i);
        }

        mViewPager.setAdapter(new CustomPagerAdapter(
                getSupportFragmentManager(), pageList, itemList));
    }

    public class CustomPagerAdapter extends FragmentPagerAdapter {

        private ArrayList<String> mPageList;
        private ArrayList<String> mItemList;
        
        public CustomPagerAdapter(FragmentManager fragmentManager
                , ArrayList<String> pageList, ArrayList<String> itemList) {
            super(fragmentManager);
            mPageList = pageList;
            mItemList = itemList;
        }

        @Override
        public Fragment getItem(int position) {
            Bundle args = new Bundle();
            args.putStringArrayList("itemList", mItemList);
            args.putInt("position", position);
            return Fragment.instantiate(getApplicationContext(), PageFragment.class.getName(), args);
        }

        @Override
        public int getCount() {
            return mPageList.size();
        }
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/banner/PagerAndListActivity.kt
package net.nend.sample.kotlin.banner

import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import androidx.appcompat.app.AppCompatActivity
import net.nend.sample.kotlin.R

/**
 * ViewPager内のListViewに表示するサンプル
 */
class PagerAndListActivity : AppCompatActivity() {

    private lateinit var viewPager: androidx.viewpager.widget.ViewPager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        viewPager = androidx.viewpager.widget.ViewPager(this)
        viewPager.id = R.id.pager
        setContentView(viewPager)

        val pageList = (1..10).map { i -> "page$i" }
        val itemList = (1..50).map { i -> "item$i" }
        viewPager.adapter = CustomPagerAdapter(supportFragmentManager, pageList, itemList)
    }

    inner class CustomPagerAdapter(
            fragmentManager: androidx.fragment.app.FragmentManager,
            private val pageList: List<String>,
            private val itemList: List<String>) : androidx.fragment.app.FragmentPagerAdapter(fragmentManager) {

        override fun getItem(position: Int): androidx.fragment.app.Fragment {
            val args = Bundle().apply {
                putStringArrayList("itemList", ArrayList(itemList))
                putInt("position", position)
            }
            return androidx.fragment.app.Fragment.instantiate(applicationContext, PageFragment::class.java.name, args)
        }

        override fun getCount() = pageList.size
    }
}